fs/remote/blob.go

```diff
@@ -187,28 +186,16 @@ func (b *blob) ReadAt(p []byte, offset int64, opts ...Option) (int, error) {
 		return 0, nil
 	}
 
-	// Make the buffer chunk aligned
-	allRegion := region{floor(offset, b.chunkSize), ceil(offset+int64(len(p))-1, b.chunkSize) - 1}
+	reg := region{offset, offset + int64(len(p)) - 1}
 	allData := make(map[region]io.Writer)
 
 	var readAtOpts options
 	for _, o := range opts {
 		o(&readAtOpts)
 	}
 
-	b.walkChunks(allRegion, func(chunk region) error {
-		var (
-			base         = positive(chunk.b - offset)
-			lowerUnread  = positive(offset - chunk.b)
-			upperUnread  = positive(chunk.e + 1 - (offset + int64(len(p))))
-			expectedSize = chunk.size() - upperUnread - lowerUnread
-		)
-
-		// Take it from remote registry.
-		// We get the whole chunk here.
-		allData[chunk] = newBytesWriter(p[base:base+expectedSize], lowerUnread)
-		return nil
-	})
+	// Take it from remote registry.
+	allData[reg] = newBytesWriter(p, 0)
```

Comment:
I don't think we need `allData` here, since there will be just a single writer now. I suggest removing it, getting writer directly and refactoring `fetchRange` to accommodate it.

Reply:
yes. I've refactored ``fetchRange`` and ``fetchRegions`` to take a single writer